### DISCUSSION TITLE
Could Not Find Macs2! Please install w/ pip,

### DISCUSSION THREAD

**Discussion by alexeykb** [GreenleafLab/ArchR#1460] [Questions / Documentation]
Hello, dear creators of ArchR.
I am running the Manual to ArchR. At the  "Chapter 10 Calling Peaks with ArchR", appears the error:

```
> pathToMacs2 <- findMacs2()
Searching For MACS2..
Not Found in $PATH
Not Found with pip
Not Found with pip3
Error in findMacs2() : 
  Could Not Find Macs2! Please install w/ pip, add to your $PATH, or just supply the macs2 path directly and avoid this function!
```

I have installed manually MACS2 via pip and provided the path to the `addReproduciblePeakSet` function
`pathToMacs2 <- "/Users/opt/anaconda3/lib/python3.9/site-packages`

Which causes the next error:

```
> pathToMacs2 <- "/Users/opt/anaconda3/lib/python3.9/site-packages"
> projHeme4 <- addReproduciblePeakSet(
+   ArchRProj = projHeme4, 
+   groupBy = "Clusters2", 
+   pathToMacs2 = pathToMacs2
+ )
Error in .checkMacs2Options(pathToMacs2) : 
  Macs2 Path (/Users/opt/anaconda3/lib/python3.9/site-packages) is out of date (version sh: /Users/opt/anaconda3/lib/python3.9/site-packages: is a directory) and does not have --shift option.
  Please update (https://github.com/taoliu/MACS) and provide new path!
In addition: Warning messages:
1: In system2(path, "callpeak -h", stdout = TRUE, stderr = TRUE) :
  running command ''/Users/opt/anaconda3/lib/python3.9/site-packages' callpeak -h 2>&1' had status 126
2: In system2(path, " --version", stdout = TRUE, stderr = TRUE) :
  running command ''/Users/opt/anaconda3/lib/python3.9/site-packages'  --version 2>&1' had status 126

```

I tried multiple options/tests and could not overpass this error.  For instance

-  I  installed the `gfortran` package and it also did not help.
- tried to use MACS3 instead of MACS2
- tried to install "`Herper`" package to further install macs2 directly from R, which did not work as well:

```


> BiocManager::install("Herper")
'getOption("repos")' replaces Bioconductor standard repositories, see '?repositories' for details

replacement repositories:
    CRAN: https://cran.rstudio.com/

Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.0 (2022-04-22)
Installing package(s) 'Herper'
Old packages: 'mnormt'
Update all/some/none? [a/s/n]: 
a

  There is a binary version available but the source version is later:
       binary source needs_compilation
mnormt  2.0.2  2.1.0              TRUE

Do you want to install from sources the package which needs compilation? (Yes/no/cancel) Yes
installing the source package ‘mnormt’

trying URL 'https://cran.rstudio.com/src/contrib/mnormt_2.1.0.tar.gz'
Content type 'application/x-gzip' length 64758 bytes (63 KB)
==================================================
downloaded 63 KB

* installing *source* package ‘mnormt’ ...
** package ‘mnormt’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
gfortran -mmacosx-version-min=10.13 -fno-optimize-sibling-calls  -fPIC  -Wall -g -O2  -c biv-nt.f -o biv-nt.o
make: gfortran: Bad CPU type in executable
make: *** [biv-nt.o] Error 1
ERROR: compilation failed for package ‘mnormt’
* removing ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/mnormt’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/mnormt’

The downloaded source packages are in
	‘/private/var/folders/1v/h8htmk4x73g6w38sj56_vb3r0000gn/T/RtmpWv1qKf/downloaded_packages’
Warning messages:
1: package ‘Herper’ is not available for Bioconductor version '3.15'

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages 
2: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
  installation of package ‘mnormt’ had non-zero exit status


```

Could you please give your thoughts about how would be better to procceed ?

Thank you
Best
Alex

ps I am using RStudio 2022.02.3 with R 4.2.0, Macintosh; Intel Mac OS X 12_3_1

**Comment by rcorces** [comment]
As denoted in the parameter definition:
https://www.archrproject.com/reference/addReproduciblePeakSet.html

<html>
<body>


pathToMacs2 | The full path to the MACS2 executable.
-- | --



</body>
</html>

Key word being the MACS2 __executable__. So you are not providing the correct input to `pathToMacs2`. For ex:
```
pathToMacs2 <- "/corces/home/rcorces/tools/python/p3.8.5/bin/macs2"
```

**Reply by alexeykb** [reply]
> As denoted in the parameter definition: https://www.archrproject.com/reference/addReproduciblePeakSet.html
> 
> pathToMacs2	The full path to the MACS2 executable.
> Key word being the MACS2 **executable**. So you are not providing the correct input to `pathToMacs2`. For ex:
> 
> ```
> pathToMacs2 <- "/corces/home/rcorces/tools/python/p3.8.5/bin/macs2"
> ```

Hello! Thank you for your reply !

I was looking for this executable file, and apparently after installation it is stored in the anaconda folder:

`pathToMacs2 <- "/Users/opt/anaconda3/bin/macs2"
`

Thank you for your fast and helpful reply! Have a great day!

**Reply by IqRaZaFaR16** [reply]
is there any possibility to  install macs2 on Rstudio via using windows system?

**Reply by immanuelazn** [reply]
Hi @IqRaZaFaR16, we do not support ArchR on windows. Thanks!

**Reply by IqRaZaFaR16** [reply]
> Hi @IqRaZaFaR16, we do not support ArchR on windows. Thanks!

Thank You
